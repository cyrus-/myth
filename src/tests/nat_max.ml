let s = Nat_decls.s ^ Compare_decls.s ^ "

let nat_max : nat -> nat -> nat |>
{
  0 => ( 0 => 0
       | 1 => 1
       | 2 => 2 )
| 1 => ( 0 => 1
       | 1 => 1
       | 2 => 2 )
| 2 => ( 0 => 2
       | 1 => 2
       | 2 => 2 )
} = ?"
