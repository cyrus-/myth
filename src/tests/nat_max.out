(* Generated Thu Nov 13 00:35:52 2014 *)
let nat_max : nat -> nat -> nat =
  let rec f1 (n1:nat) : nat -> nat =
    fun (n2:nat) ->
      match n1 with
        | O -> n2
        | S (n3) -> (match n2 with
                       | O -> n1
                       | S (n4) -> S (f1 n3 n4))
  in
    f1
;;
